drivers/src/stm32f446xx_usart_driver_course.c

```diff
@@ -6,16 +6,86 @@
  */
 
 #include "stm32f446xx_usart_driver_course.h"
+#include "stm32f446xx_rcc.h"
+
+void UART_PeriClockControl(USART_RegDef_t* pUSARTx, uint8_t EnableDisable)
+{
+
+	if(EnableDisable == ENABLE)
+	{
+		if(pUSARTx == USART1)
+		{
+			USART1_PCLK_EN();
+		}
+		else if(pUSARTx == USART2)
+		{
+			USART2_PCLK_EN();
+		}
+		else if(pUSARTx == USART3)
+		{
+			USART3_PCLK_EN();
+		}
+		else if(pUSARTx == USART4)
+		{
+			UART4_PCLK_EN();
+		}
+		else if(pUSARTx == USART5)
+		{
+			UART5_PCLK_EN();
+		}
+		else if(pUSARTx == USART6)
+		{
+			USART6_PCLK_EN();
+		}
+	}
+	else if(EnableDisable == DISABLE)
+	{
+		if(pUSARTx == USART1)
+		{
+			USART1_PCLK_DI();
+		}
+		else if(pUSARTx == USART2)
+		{
+			USART2_PCLK_DI();
+		}
+		else if(pUSARTx == USART3)
+		{
+			USART3_PCLK_DI();
+		}
+		else if(pUSARTx == USART4)
+		{
+			UART4_PCLK_DI();
+		}
+		else if(pUSARTx == USART5)
+		{
+			UART5_PCLK_DI();
+		}
+		else if(pUSARTx == USART6)
+		{
+			USART6_PCLK_DI();
+		}
+	}
+}
 
 /* USART Initialization */
 void USART_Initialize(USART_Handler_TypedDef* pUSART_Handler)
 {
 	//baud rate equation : txrx baud = Fclk/(8*(2-oversampling)*USARTDIV)
 	float USARTDIV, SysClock;
 
-//	USARTDIV = f/((8*2-pUSART_Handler->USART_Pin_Config.USART_OverSampling)*pUSART_Handler->USART_Pin_Config.USART_BaudRate);
+	/* Enable UART */
+	pUSART_Handler->pUSART->CR1 |= (1ul << 13);
```

Comment:
Is way easier if you give names (the ones from the Reference Manual) from the Bit positiion

#define USART_CR1_UE 13
pUSART_Handler->pUSART->CR1 |= (1ul << USART_CR1_UE);